File: 26.remove_duplicates.py
from typing import List


class Solution:
    """
    https://leetcode.com/problems/remove-duplicates-from-sorted-array/
    """
    def removeDuplicates(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return 0
        left = 1
        for right in range(1, len(nums)):
            if nums[right] != nums[right - 1]:
                nums[left] = nums[right]
                left += 1
        return left

File: 26. removeDuplicates.ts
// https://leetcode.com/problems/remove-duplicates-from-sorted-array
function removeDuplicates(nums: number[]): number {
    if (nums.length === 0) {
        return 0
    }
    let counter = 1
    for (let i = 1; i < nums.length; i++) {
        if (nums[i] !== nums[i - 1]) {
            nums[counter] = nums[i]
            counter++
        }
    }
    return counter
};
